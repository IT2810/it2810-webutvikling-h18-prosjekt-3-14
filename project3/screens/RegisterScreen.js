import React from 'react';
import Expo from "expo";
import { View, Button, AsyncStorage, StyleSheet, Text, TextInput, Picker, ImageBackground, TouchableOpacity } from 'react-native';
 
export default class RegisterScreen extends React.Component {
  static navigationOptions = {
    title: 'Register screen',
  };
  
  constructor(props){
      super(props);
      this.state = {
        name: '',
        gender: 'male',
        age: 0,
        weight: 0,
        height: 0,
        goal: 0,
      }
    }

    //Function returning true if the required user info is entered.
    enableButton() {
      return !(this.state.name.length > 0 && this.state.age > 0 && this.state.weight > 0 && this.state.height > 0 && this.state.goal > 0);
    }
 
    render() {
      return (
          <View style={styles.inputcontainer}>
              <Text style={styles.label}> {this.state.gender === 'male' ? "ü§µ" : "üë©"} Name </Text>
              <TextInput
                label="Name"
                style={styles.input}
                onChangeText={(text) => this.setState({name: text})}
                placeholder="Jon Doe"
                textContentType="name"
                enablesReturnKeyAutomatically={true}
                returnKeyType="next"
                value={this.state.name}
              />
              <Text style={styles.label}>üî¢ Age</Text>
              <TextInput
                style={styles.input}
                onChangeText={(text) => this.setState({age: text})}
                placeholder="21"
                value={this.state.age.toString()}
                keyboardType="number-pad"
                returnKeyType="done"
              />
              <Text style={styles.label}>{this.state.gender === 'male' ? "üèãÔ∏è‚Äç":"üèãÔ∏è‚Äç‚ôÄÔ∏è"} Weight </Text>
              <TextInput
                style={styles.input}
                onChangeText={(text) => this.setState({weight: text})}
                placeholder="79,0 (kg)"
                value={this.state.weight.toString()}
                keyboardType="numeric"
                returnKeyType="done"
              />
              <Text style={styles.label}>üìè Height</Text>
              <TextInput
                style={styles.input}
                onChangeText={(text) => this.setState({height: text})}
                placeholder="182,5 (cm)"
                value={this.state.height.toString()}
                keyboardType="numeric"
                returnKeyType="done"
              />
 
              <Text style={styles.label}> üèÜ Goal </Text>
              <TextInput
                style={styles.input}
                onChangeText={(text) => this.setState({goal: text})}
                placeholder="Enter your target step count per day"
                value={this.state.goal.toString()}
                keyboardType="number-pad"
                returnKeyType="done"
              />
              <Text style={styles.label}>Gender</Text>
              <View style={styles.genderButtonsContainer}>
                <TouchableOpacity style={(this.state.gender === 'male') ? styles.activebutton : styles.inactivebutton} onPress={() => this.setState({gender: 'male'})}>
                  <View style={{alignItems: "center"}}>
                    <Text style={styles.buttonText}>
                    üèÉ‚Äç‚ôÇÔ∏è
                    </Text>
                  </View>
                </TouchableOpacity>
                <TouchableOpacity style={(this.state.gender === 'female') ? styles.activebuttonpink : styles.inactivebutton} onPress={() => this.setState({gender: 'female'})}>
                  <View style={{alignItems: "center"}}>
                    <Text style={styles.buttonText}>
                    üèÉ‚Äç‚ôÄÔ∏è
                    </Text>
                  </View>
                </TouchableOpacity>
              </View>
              <View style={styles.buttoncontainer}>
                <Button title="Submit" onPress={this._signInAsync} color="rgb(28, 184, 65)" disabled={this.enableButton()}/>
              </View>
          </View>  
      );
    }
 
    //Function beeing called when submitting user information -> Storing info in AsyncStorage and navigating to App.
    _signInAsync = async () => {
      let USER_object = {
        name: this.state.name,
        age: this.state.age,
        gender: this.state.gender,
        weight: this.state.weight,
        height: this.state.height,
        goal: this.state.goal,
      };
      await AsyncStorage.setItem('USER', JSON.stringify(USER_object));
      this.props.navigation.navigate('App');
    };
 
  }
 
  const styles = StyleSheet.create({
    inputcontainer: {
      paddingTop: 20,
      alignItems: "center",
      flex:1,
      backgroundColor: "#F8F8F8"
    },
    input: {
      height: "3.7%",
      width: "85%",
      textAlign: "center",
      fontSize: 14,
      borderLeftWidth: 1,
      borderTopWidth: 1,
      borderRightWidth: 1,
      borderBottomWidth: 1,
      borderTopLeftRadius: 50,
      borderTopRightRadius: 7,
      borderBottomRightRadius: 50,
      borderBottomLeftRadius: 7,
      borderColor: "dodgerblue",
      backgroundColor: "white",
    },
    label: {
      color: "dodgerblue",
      fontSize: 23,
      marginTop: 14,
      marginBottom: 1
    },
    buttoncontainer: {
      marginTop: 15
    },
    activebutton: {
      backgroundColor: '#6F9FD8',
      padding: 2,
      width: "40%",
      borderTopLeftRadius: 50,
      borderTopRightRadius: 7,
      borderBottomRightRadius: 50,
      borderBottomLeftRadius: 7,
      marginRight: 8
    },
    activebuttonpink: {
      backgroundColor: '#FF69B4',
      padding: 2,
      width: "40%",
      borderTopLeftRadius: 50,
      borderTopRightRadius: 7,
      borderBottomRightRadius: 50,
      borderBottomLeftRadius: 7,
      marginRight: 8,
      justifyContent: "center"    
    },
    inactivebutton: {
      backgroundColor: 'grey',
      padding: 2,
      width: "40%",
      borderTopLeftRadius: 50,
      borderTopRightRadius: 7,
      borderBottomRightRadius: 50,
      borderBottomLeftRadius: 7,
      marginRight: 8,
 
    },
    genderButtonsContainer: {
      marginTop: 10,
      flexDirection: 'row',
      justifyContent: 'center',
    },
    buttonText: {
      fontSize: 30,
      color: "white",
    }
  });
 
  Expo.registerRootComponent(RegisterScreen);
