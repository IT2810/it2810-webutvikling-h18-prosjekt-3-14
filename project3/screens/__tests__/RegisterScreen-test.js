import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import RegisterScreen from "../RegisterScreen";
it("PedometerScreen Matches snapshot.", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<RegisterScreen/>);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
});