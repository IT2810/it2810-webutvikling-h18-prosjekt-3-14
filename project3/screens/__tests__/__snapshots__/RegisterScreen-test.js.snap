// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PedometerScreen Matches snapshot. 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F8F8F8",
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "dodgerblue",
        "fontSize": 23,
        "marginBottom": 1,
        "marginTop": 14,
      }
    }
  >
     
    🤵
     Name 
  </Text>
  <TextInput
    allowFontScaling={true}
    enablesReturnKeyAutomatically={true}
    label="Name"
    onChangeText={[Function]}
    placeholder="Jon Doe"
    returnKeyType="next"
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomLeftRadius": 7,
        "borderBottomRightRadius": 50,
        "borderBottomWidth": 1,
        "borderColor": "dodgerblue",
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "borderTopLeftRadius": 50,
        "borderTopRightRadius": 7,
        "borderTopWidth": 1,
        "fontSize": 14,
        "height": "3.7%",
        "textAlign": "center",
        "width": "85%",
      }
    }
    textContentType="name"
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "dodgerblue",
        "fontSize": 23,
        "marginBottom": 1,
        "marginTop": 14,
      }
    }
  >
    🔢 Age
  </Text>
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    onChangeText={[Function]}
    placeholder="21"
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomLeftRadius": 7,
        "borderBottomRightRadius": 50,
        "borderBottomWidth": 1,
        "borderColor": "dodgerblue",
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "borderTopLeftRadius": 50,
        "borderTopRightRadius": 7,
        "borderTopWidth": 1,
        "fontSize": 14,
        "height": "3.7%",
        "textAlign": "center",
        "width": "85%",
      }
    }
    value={0}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "dodgerblue",
        "fontSize": 23,
        "marginBottom": 1,
        "marginTop": 14,
      }
    }
  >
    🏋️‍
     Weight 
  </Text>
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="79,0 (kg)"
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomLeftRadius": 7,
        "borderBottomRightRadius": 50,
        "borderBottomWidth": 1,
        "borderColor": "dodgerblue",
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "borderTopLeftRadius": 50,
        "borderTopRightRadius": 7,
        "borderTopWidth": 1,
        "fontSize": 14,
        "height": "3.7%",
        "textAlign": "center",
        "width": "85%",
      }
    }
    value={0}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "dodgerblue",
        "fontSize": 23,
        "marginBottom": 1,
        "marginTop": 14,
      }
    }
  >
    📏 Height
  </Text>
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="182,5 (cm)"
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomLeftRadius": 7,
        "borderBottomRightRadius": 50,
        "borderBottomWidth": 1,
        "borderColor": "dodgerblue",
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "borderTopLeftRadius": 50,
        "borderTopRightRadius": 7,
        "borderTopWidth": 1,
        "fontSize": 14,
        "height": "3.7%",
        "textAlign": "center",
        "width": "85%",
      }
    }
    value={0}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "dodgerblue",
        "fontSize": 23,
        "marginBottom": 1,
        "marginTop": 14,
      }
    }
  >
     🏆 Goal 
  </Text>
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    onChangeText={[Function]}
    placeholder="Enter your target step count per day"
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomLeftRadius": 7,
        "borderBottomRightRadius": 50,
        "borderBottomWidth": 1,
        "borderColor": "dodgerblue",
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "borderTopLeftRadius": 50,
        "borderTopRightRadius": 7,
        "borderTopWidth": 1,
        "fontSize": 14,
        "height": "3.7%",
        "textAlign": "center",
        "width": "85%",
      }
    }
    value={0}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "dodgerblue",
        "fontSize": 23,
        "marginBottom": 1,
        "marginTop": 14,
      }
    }
  >
    Gender
  </Text>
  <Component
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 10,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#6F9FD8",
          "borderBottomLeftRadius": 7,
          "borderBottomRightRadius": 50,
          "borderTopLeftRadius": 50,
          "borderTopRightRadius": 7,
          "marginRight": 8,
          "padding": 2,
          "width": "40%",
        }
      }
    >
      <Component
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 30,
            }
          }
        >
          🏃‍♂️
        </Text>
      </Component>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
          "borderBottomLeftRadius": 7,
          "borderBottomRightRadius": 50,
          "borderTopLeftRadius": 50,
          "borderTopRightRadius": 7,
          "marginRight": 8,
          "padding": 2,
          "width": "40%",
        }
      }
    >
      <Component
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 30,
            }
          }
        >
          🏃‍♀️
        </Text>
      </Component>
    </TouchableOpacity>
  </Component>
  <Component
    style={
      Object {
        "marginTop": 15,
      }
    }
  >
    <Button
      color="rgb(28, 184, 65)"
      disabled={true}
      onPress={[Function]}
      title="Submit"
    />
  </Component>
</Component>
`;
